src/main/java/beam/physsim/jdeqsim/cacc/sim/Road.java

```diff
@@ -172,7 +172,7 @@ private double  getNextAvailableTimeForLeavingStreet(double simTime){
     public void procsessFilledRoad(org.matsim.core.mobsim.jdeqsim.Vehicle vehicle, double simTime) {
         LinkedList<Double> gap = getGap();
         if (gap == null) {
-            gap = new LinkedList<>();
+            setGap(new LinkedList<>());
```

Comment:
`setGap` will set the class field, but here was no such intention. I guess it is better not to do it

Reply:
The variable gap was not used after assignment. So either was redundant assignment or the intention was to set the field.

Reply:
I agree with Carlos - that seems to be a bug to me. Since it's a difference of null versus empty it likely didn't make a difference - if they are treated the same.

Reply:
Yes, Gap is redundent. We can remove it. Only intention was if we use gap in method for further processing in future then we find not null value of gap

Reply:
Also keep this gap.clear() with null check. because we are clearing gap value in this method